crypto/sm2/sm2_crypt.c

```diff
@@ -67,6 +67,17 @@ static size_t ec_field_size(const EC_GROUP *group)
     return field_size;
 }
 
+static int is_all_zeros(const unsigned char *msg, size_t msglen)
+{
+    unsigned char re = 0;
+    size_t i;
+
+    for (i = 0; i < msglen; i++) {
+        re |= msg[i];
+    }
+    return re == 0 ? 1 : 0;
+}
+
```

Comment:
This might not be worth doing, but you could use ffsl() here and scan the msg in long word chunks rather than by individual bytes, which might be faster

Reply:
`ffsl()` takes a `long int` - I haven't looked at the code, but if it can cast the `unsigned char *` to `long int *` without worrying about alignment it could just check for `0L`

However, I suspect that checking this value for all zeroes isn't going to be the bottleneck here

Reply:
Yeah, this would be a premature optimization IMO.